bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java

```diff
@@ -126,7 +126,10 @@ public ShellyDeviceProfile initialize(String thingType, String jsonIn) throws Sh
         initFromThingType(thingType);
 
         String json = jsonIn;
-        if (json.contains("\"ext_temperature\":{\"0\":[{")) {
+        // It is not guaranteed, that the array entries are in order. Check all
+        // possible variants. See openhab#15514.
+        if (json.contains("\"ext_temperature\":{\"0\":[{") || json.contains("\"ext_temperature\":{\"1\":[{")
+                || json.contains("\"ext_temperature\":{\"2\":[{")) {
```

Comment:
Wouldn't this be sufficient ?
```
if (json.contains("\"ext_temperature\":{\"")) {
```

Reply:
I can't tell.

I assume there was a reason, the JSON string is checked including the array index. So I just added the two other possible cases, but keeping as close to the original condition as possible.

Reply:
Let's wait for @markus7017 feedback